Firebase/InstanceID/FIRInstanceID.m

```diff
@@ -1092,4 +1098,35 @@ - (void)logAPNSConfigurationError:(NSString *)errorString {
   }
 }
 
+#pragma mark - Sync InstanceID
+
+- (void)updateFirebaseInstallationID {
+  FIRInstanceID_WEAKIFY(self);
+  [self.installations
+      installationIDWithCompletion:^(NSString *_Nullable installationID, NSError *_Nullable error) {
+        FIRInstanceID_STRONGIFY(self);
+        self.firebaseInstallationsID = installationID;
```

Comment:
This seems set async, so does it mean there's a chance that if analytics query it too fast, it might be nil?

Reply:
Yes, there is a chance that it will be `nil`. Particularly for analytics it will unlikely be an issue due to their use case - IID is requested before sending data which happens way after Firebase initialization (we discussed it with @htcgh some time ago). 

The sync method is deprecated in favour the public async method and will be deleted eventually. Taking this into account it looks better to me to provide safe and fast implementation which may return `nil` (which was a case even before) rather than keep slow and unsafe (may lead to a deadlock).

AFIK, the sync method is not be used by recent versions of other SDKs (except Analytics), so it is mostly to be on the safe side in the case when e.g. an old version of remote config is locked but IID is update. 

WDYT?

Reply:
I wonder if we could just get rid of the sync method and ask analytics to call the async method instead. Otherwise we are adding extra code just to call async method from a sync one.

Reply:
Yeah, actually you're right. Removing the method will require a minor version bump for all dependent SDKs, but it is what we will do at the release time anyway. This is something we'll need to discuss with them though. BTW, Analytics will be doing it eventually. The implementation will be very similar on the Analytics side (cl/262474711).

Reply:
Nope, we can't delete private APIs from IID until a major version update since locking to an old 6.x version of Analytics allows IID minor version updates of its IID dependency.

Reply:
@paulb777 Yeah, good point, there are some valid combinations of locks when the removed API may be called.

@chliangGoogle I created a bug #4486 to remove this code for the Firebase 7 release.

Reply:
I'm okay with current change in that case. maybe we can add a todo here to say when major update, we can just get rid of this method?

Reply:
This is a special case for Analytics or just in general we can't remove private API until major version update?

Reply:
It's technically true for any private API used across pods in Firebase.

It's especially true for IID since that doesn't get included and locked with the Firebase pod.